docs/body_file.md

```diff
@@ -8,138 +8,156 @@ tags:
+* For hard links on HFS+ the Catalog Node Identifier (CNID) of the link target (indirect node) file record is used instead as the `inode` value instead of the CNID of the (hard link) file record itself. This matches the behavior of Mac OS (file) stat as described [here](https://developer.apple.com/library/archive/technotes/tn/tn1150.html), in the section "Hard Links".
+* For HFS+ the MD5 calculation of `fls` includes:
+  * Regular files
+  * symbolic links (content of the data stream of the symbolic link not its target)
+  * "Virtual metadata files" like `$CatalogFile`
+
+### EXT2
+
+* For ext2 the MD5 calculation of `fls` includes:
+  * Regular files
+  * Directories (contents of the directory entries data stream)
+  * Symbolic links (content of the data stream of the symbolic link not its
+  target)
+  * "Virtual metadata files/directories" like `$OrphanFiles`
+
+## Output Format
+
+Sleuthkit 4.7.0 was used to create the example below from a sample NTFS MFT file entry:
+
+Body file data:
+
+```ascii
+0|/$BadClus|8-128-2|r/rr-xr-xr-x|0|0|0|1580550524|1580550524|1580550524|1580550524
+0|/$BadClus:$Bad|8-128-1|r/rr-xr-xr-x|0|0|7270400|1580550524|1580550524|1580550524|1580550524
+0|/$BadClus ($FILE_NAME)|8-48-3|r/rr-xr-xr-x|0|0|82|1580550524|1580550524|1580550524|1580550524
+```
+
+Corresponding NTFS MFT entry:
+
+```ascii
+MFT Entry Header Values:
+Entry: 8        Sequence: 8
+$LogFile Sequence Number: 1069751
+Allocated File
+Links: 1
+
+$STANDARD_INFORMATION Attribute Values:
+Flags: Hidden, System
+Owner ID: 0
+Security ID: 256  ()
+Created:    2020-02-01 10:48:44.857384500 (CET)
+File Modified:  2020-02-01 10:48:44.857384500 (CET)
+MFT Modified:   2020-02-01 10:48:44.857384500 (CET)
+Accessed:   2020-02-01 10:48:44.857384500 (CET)
+
+$FILE_NAME Attribute Values:
+Flags: Hidden, System
+Name: $BadClus
+Parent MFT Entry: 5     Sequence: 5
+Allocated Size: 0       Actual Size: 0
+Created:    2020-02-01 10:48:44.857384500 (CET)
+File Modified:  2020-02-01 10:48:44.857384500 (CET)
+MFT Modified:   2020-02-01 10:48:44.857384500 (CET)
+Accessed:   2020-02-01 10:48:44.857384500 (CET)
+
+Attributes: 
+Type: $STANDARD_INFORMATION (16-0)   Name: N/A   Resident   size: 72
+Type: $FILE_NAME (48-3)   Name: N/A   Resident   size: 82
+Type: $DATA (128-2)   Name: N/A   Resident   size: 0
+Type: $DATA (128-1)   Name: $Bad   Non-Resident   size: 7270400  init_size: 0
+```
 
 Behavior of the TSK Metadata Address:
 
-- 11-144-4 references the \$INDEX_ROOT attribute in MFT entry 11, note
-  that this \$INDEX_ROOT attribute on-disk is the 3rd attribute in the
-  MFT entry. Note that the size of a corresponding \$INDEX_ALLOCATION is
-  not included in the "size" in the body file.
-  [4](https://github.com/sleuthkit/sleuthkit/issues/1831)
+* `8-128-2` references the `$DATA` attribute in MFT entry 8, note that this
```

Comment:
Add a note that this breaks down when there is an ATTRIBUTE_LIST since it is not deterministic what the "first" MFT attribute is.

Reply:
Addressed in commit.